### PR TITLE
Update documentation for Feathr UI, registry, and architecture.

### PR DESCRIPTION
**Pull request by xiaoyongzhu**
This PR focus on:
- Added the Feature registry doc (see https://github.com/linkedin/feathr/blob/xiaoyzhu/release_update_and_registry/docs/concepts/feature-registry.md for the rendering)
- Highlight Feathr UI
- updated the architecture diagram (see PR files for the new architecture) to reflect the latest change.
- formatted the ARM deployment doc
- typo fix

Solved doc issue:

https://github.com/linkedin/feathr/issues/493
https://github.com/linkedin/feathr/issues/292

### CODE REVIEW

File: docs/concepts/feature-registry.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+```yaml
+feature_registry:
+  # The API endpoint of the registry service
+  api_endpoint: "https://feathr-sql-registry.azurewebsites.net/api/v1"
+```
+
+### Register and Listing Features
+
+You can register your features in the centralized registry and share it with other team members who want to consume those features and for further use. You can also use `list_registered_features` to verify if they have been registered successfully.
```
Comment (Yuqing-cat) [inline docs/concepts/feature-registry.md:65]: Do we have a use case for feature sharing?

File: docs/how-to-guides/azure-deployment-arm.md [hunk 2]
```diff
@@ -28,53 +28,78 @@ The architecture diagram demonstrates how different Azure components interact wi
...
-   #Fetch the objectId of AAD app to patch it and add redirect URI in next step.
-   aad_objectId=$(az ad app list --display-name $sitename --query [].id -o tsv)
+### 3. Grant Key Vault and Synapse access to selected users (Optional)
 
-   # Make sure the above command ran successfully and the values are not empty. If they are empty, re-run the above commands as the app creation could take some time.
-   # MAKE NOTE OF THE CLIENT_ID & TENANT_ID FOR STEP #2
-   echo "AZURE_AAD_OBJECT_ID: $aad_objectId"
-   echo "AAD_CLIENT_ID: $aad_clientId"
-   echo "AZURE_TENANT_ID: $aad_tenantId"
+YOu will need to assign the right permission to users in order for them to access Azure key vault, permission to access the Storage Blob as a Contributor, and permission to submit jobs to Synapse cluster.
```
Comment (Yuqing-cat) [inline docs/how-to-guides/azure-deployment-arm.md:80]: Typo, `YOu` -> `You`

File: docs/quickstart_synapse.md [hunk 3]
```diff
@@ -24,9 +24,8 @@ Feathr has native cloud integration. Here are the steps to use Feathr on Azure:
 
 1. Follow the [Feathr ARM deployment guide](https://linkedin.github.io/feathr/how-to-guides/azure-deployment-arm.html) to run Feathr on Azure. This allows you to quickly get started with automated deployment using Azure Resource Manager template. Alternatively, if you want to set up everything manually, you can checkout the [Feathr CLI deployment guide](https://linkedin.github.io/feathr/how-to-guides/azure-deployment-cli.html) to run Feathr on Azure. This allows you to understand what is going on and set up one resource at a time.
 
-2. Once the deployment is complete,run the Feathr Jupyter Notebook by clicking the button below. You only need to change the specified `Resource Prefix`.
-
-[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/linkedin/feathr/main?labpath=feathr_project%2Ffeathrcli%2Fdata%2Ffeathr_user_workspace%2Fnyc_driver_demo.ipynb)
+2. Once the deployment is complete,run the Feathr Jupyter Notebook by clicking this button:  [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/linkedin/feathr/main?labpath=feathr_project%2Ffeathrcli%2Fdata%2Ffeathr_user_workspace%2Fnyc_driver_demo.ipynb).
```
Comment (Yuqing-cat) [inline docs/quickstart_synapse.md:27]: The Binder link works for me this time. We might be able to close #359 if no other issues reported.

Comment (xiaoyongzhu) [reply docs/quickstart_synapse.md:27]: yes https://github.com/linkedin/feathr/pull/520 should have fixed this.

File: docs/concepts/feature-registry.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+## Deployment Options
+
+Feathr supports two types of backends for Feature Registry - Azure Purview (Apache Atlas compatible service) and ANSI SQL. Depending on your IT setup, you might choose either of those in the above deployment steps.
+
+Note that if you choose to enable Role-based Access Control (RBAC), you still need to use a SQL service (such as Azure SQL) to store all the RBAC related information.
+
+## Architecture
+
+![Architecture Diagram](../images/architecture.png)
```
[the remark this answers is not in the document]
Comment (xiaoyongzhu) [reply docs/concepts/feature-registry.md:41]: good point!

File: docs/concepts/feature-registry.md (hunk 1 above)
Comment (xiaoyongzhu) [reply docs/concepts/feature-registry.md:65]: yes and I make that part explicit. Basically just tell the feature consumers on which features can be consumed.

**Review by Yuqing-cat — APPROVED** [review]

File: docs/concepts/feature-registry.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+layout: default
+title: Feature Registry
+parent: Feathr Concepts
+---
+
+# Feature Registry and Feathr UI
+
+Feature registry is an important component of a feature store. This documentation will cover the supported backend of a feature registry and the usages.
+
+## Introduction
```
Comment (blrchen) [inline docs/concepts/feature-registry.md:11]: Can we add some content under this section:

You can use Feathr UI to 
- Identity Data Sources for features
- Search features and explore feature metadata details
- Track project and feature level lineage data
- Manage access control

**Review by blrchen — APPROVED** [review]
Thanks for the doc improvements, PR looks good to me, only one minor comment

**Review by Yuqing-cat — APPROVED** [review]

**Review by blrchen — APPROVED** [review]

**Pull request merged**